tests/mock_crust/churn.rs

```diff
@@ -51,126 +53,136 @@ fn random_churn<R: Rng>(rng: &mut R,
+type GetKey = (DataIdentifier, MessageId, Authority, Authority);
 
-// Check that the given node received a Get request with the given details.
-fn did_receive_get_request(node: &TestNode,
-                           expected_src: Authority,
-                           expected_dst: Authority,
-                           expected_data_id: DataIdentifier,
-                           expected_message_id: MessageId)
-                           -> bool {
-    loop {
-        match node.event_rx.try_recv() {
-            Ok(Event::Request { request: Request::Get(data_id, message_id), ref src, ref dst })
-                if *src == expected_src && *dst == expected_dst && data_id == expected_data_id &&
-                   message_id == expected_message_id => return true,
-            Ok(_) => (),
-            Err(_) => return false,
-        }
-    }
+/// A set of expectations: Which nodes and groups are supposed to receive Get requests.
+#[derive(Default)]
+struct ExpectedGets {
+    /// The Get requests expected to be received.
+    messages: HashSet<GetKey>,
+    /// The group members of the receiving groups, at the time of sending.
+    groups: HashMap<Authority, HashSet<XorName>>,
 }
 
-fn did_receive_get_success(node: &TestNode,
-                           expected_src: Authority,
-                           expected_dst: Authority,
-                           expected_data: Data,
-                           expected_message_id: MessageId)
-                           -> bool {
-    loop {
-        let expected = |src: &Authority, dst: &Authority, data: &Data, message_id: MessageId| {
-            *src == expected_src && *dst == expected_dst && *data == expected_data &&
-            message_id == expected_message_id
-        };
-        match node.event_rx.try_recv() {
-            Ok(Event::Response { response: Response::GetSuccess(ref data, message_id),
-                                 ref src,
-                                 ref dst }) if expected(src, dst, data, message_id) => return true,
-            Ok(_) => (),
-            Err(_) => return false,
+impl ExpectedGets {
+    /// Sends a request using the nodes specified by `src`, and adds the expectation that the nodes
+    /// belonging to `dst` receive the message. Panics if not enough nodes sent a group message, or
+    /// if an individual sending node could not be found.
+    fn send_and_expect(&mut self,
+                       data_id: DataIdentifier,
+                       src: Authority,
+                       dst: Authority,
+                       nodes: &[TestNode]) {
+        let src_dest = src.to_destination();
+        let dst_dest = dst.to_destination();
+        let msg_id = MessageId::new();
+        let mut sent_count = 0;
+        for node in nodes.iter().filter(|node| node.is_recipient(&src_dest)) {
+            let result = node.inner.send_get_request(src, dst, data_id, msg_id);
```

Comment:
I suppose this could be reduced to just `unwrap!(node.inner.send_get_request(src, dst, data_id, msg_id));` to avoid the explicit `assert!` on the next line.